2/contrib/s2i/run

```diff
@@ -235,8 +277,39 @@ if [ "$(ls /usr/lib/jenkins/*.hpi 2>/dev/null)" ]; then
           continue
         fi
 	echo "Linking ${FILENAME} RPM installed Jenkins plugins to ${JENKINS_HOME} ..."  
+        # note note a force link, to not override upgrades when not upgrading the image
         ln -s  $FILENAME ${JENKINS_HOME}/plugins/${basefilename}.jpi
    done
+   
+   # since this is rhel, if it is also the initial start of the image, we can remove the 
+   # the /usr/lib/jenkins/*.hpi link from /opt/openshift/plugins/*.jpi that 
+   # was done in the rhel7 dockefile; that link is done to assist our s2i
+   # extension scenario, so that the existing plugins are available during 
+   # the s2i assemble step (which precedes the s2i run step handled by this
+   # script); however, we must keep any plugins pulled in by s2i assemble, so
+   # we only remove files which are the same as /usr/lib/jenkins
+   #
+   # also, this will preserve the links if nothing has changed so that subsequent
+   # release to release migrations can work since the links will result in picking
+   # up the new content in /usr/lib/jenkins
+   if [ ! -e ${JENKINS_HOME}/configured ]; then
+      echo "Cleaning up links to base image plugins in /opt/openshift/plugins as part of this initial image start"
+      for FILENAME in /opt/openshift/plugins/* ; do
+         basefilename=`basename $FILENAME .jpi`
+         rpm_plugin_location="/usr/lib/jenkins/${basefilename}.hpi"
+         echo "Looking at ${FILENAME} and ${rpm_plugin_location}"
+         if [ "$(ls ${rpm_plugin_location} 2>/dev/null)" ]; then
+            optcksum=`cksum ${FILENAME}`
+            rpmcksum=`cksum ${rpm_plugin_location}`
+            echo "${FILENAME} cksum ${optcksum} and ${rpm_plugin_location} cksum ${rpmcksum}"
+            echo "Removing ${FILENAME}"
+            rm -f $FILENAME
+         fi
+      done
+      if [ "$(ls /opt/openshift/plugins/* 2>/dev/null)" ]; then
+         echo "Remaining plugins in /opt/openshift/plugins are $(ls /opt/openshift/plugins)"
+      fi
+   fi
 fi
 
 if [ ! -e ${JENKINS_HOME}/configured ]; then
```

Comment:
why didn't you just put the above block inside here, rather than introducing a second "if not configured" block?

Reply:
Could have done that yeah, but would have had to duplicate the `if [ "$(ls /usr/lib/jenkins/*.hpi 2>/dev/null)" ]; then` if check that this is currently in.

Seems like a net zero gain to me.

Reply:
ah, yep, missed the additional outer if block.